Added a verification procedure for LgtmScore

artem-smotrakov opened the pull request:
Here is a list of updates:

- Added a couple of checks in the `LgtmScore` class.
- Defined test vectors for the `LgtmScore`.
- Added a new `LgtmScoreVerification` class.
- Updated the `LgtmScoreTest`.

This is a part of #24

sourabhsparkala commented on src/main/java/com/sap/sgs/phosphor/fosstars/model/score/oss/LgtmScore.java:
```diff
@@ -31,14 +36,14 @@
     GRADE_TO_POINTS.put(LgtmGrade.B, 6.0);
```
Any reason for A_PLUS and A having the same Point?

artem-smotrakov replied:
I think they both are good and deserve the same number of points.

sourabhsparkala commented on src/main/java/com/sap/sgs/phosphor/fosstars/model/score/oss/LgtmScore.java:
```diff
@@ -49,11 +54,19 @@ public ScoreValue calculate(Value... values) {
 
+    if (usesLgtm.isUnknown() && !worseLgtmGrade.isUnknown()) {
+      throw new IllegalArgumentException(
+          "Hey! It's unknown if the project uses LGTM but then you gave me the worse grade!");
```
Please change the end of the sentence to "gave me the worst grade!"

artem-smotrakov replied:
Then, I think "worse" needs to be replaced with "worst" everywhere.

sourabhsparkala commented on src/main/java/com/sap/sgs/phosphor/fosstars/model/score/oss/LgtmScore.java:
```diff
@@ -49,11 +54,19 @@ public ScoreValue calculate(Value... values) {
         scoreValue.increase(LGTM_USAGE_POINTS);
+      } else if (!worseLgtmGrade.isUnknown()) {
+        throw new IllegalArgumentException(
+            "Hey! You told me that LGTM is not used but then provided the worse grade!");
```
Please change the end of the sentence to "provided the worst grade!"

sourabhsparkala commented on src/main/resources/com/sap/sgs/phosphor/fosstars/model/score/oss/LgtmScoreTestVectors.csv:
```diff
@@ -0,0 +1,9 @@
+4,test_vector_4,5.0,7.0,,True,C
+5,test_vector_5,7.0,9.0,,True,B
+6,test_vector_6,9.0,10.0,,True,A
+7,test_vector_7,9.0,10.0,,True,A+
```
Difference between A and A+, is not clear in this test_vector

artem-smotrakov replied:
There should not be any difference for now since the score considers them equal.

artem-smotrakov commented:
@sourabhsparkala I have renamed the `WORSE_LGTM_GRADE` feature to `WORST_LGTM_GRADE` and also changed all related strings. Good catch! Thanks for the review!
